tools/migrators/lib/opsgenie/resources/integrations.py

```diff
@@ -0,0 +1,63 @@
+import re
+from typing import List
+
+from lib.oncall.api_client import OnCallAPIClient
+from lib.opsgenie.config import (
+    OPSGENIE_FILTER_INTEGRATION_REGEX,
+    OPSGENIE_FILTER_TEAM,
+    OPSGENIE_TO_ONCALL_VENDOR_MAP,
+    UNSUPPORTED_INTEGRATION_TO_WEBHOOKS,
+)
+
+
+def filter_integrations(integrations: list[dict]) -> list[dict]:
+    """Apply filters to integrations."""
+    if OPSGENIE_FILTER_TEAM:
+        integrations = [
+            i for i in integrations if i.get("teamId") == OPSGENIE_FILTER_TEAM
+        ]
+
+    if OPSGENIE_FILTER_INTEGRATION_REGEX:
+        pattern = re.compile(OPSGENIE_FILTER_INTEGRATION_REGEX)
+        integrations = [i for i in integrations if pattern.match(i["name"])]
+
+    return integrations
+
+
+def match_integration(integration: dict, oncall_integrations: List[dict]) -> None:
+    """
+    Match OpsGenie integration with Grafana OnCall integration + match opsgenie
+    integration type with Grafana OnCall integration type.
+    """
+    oncall_integration = None
+    for candidate in oncall_integrations:
+        name = integration["name"].lower().strip()
+        if name == candidate["name"].lower().strip():
+            oncall_integration = candidate
+
+    integration["oncall_integration"] = oncall_integration
+
+    integration_type = OPSGENIE_TO_ONCALL_VENDOR_MAP.get(integration["type"])
+    if not integration_type and UNSUPPORTED_INTEGRATION_TO_WEBHOOKS:
+        integration_type = "webhook"
+    integration["oncall_type"] = integration_type
+
+
+def migrate_integration(integration: dict) -> None:
+    """Migrate OpsGenie integration to Grafana OnCall."""
+    if integration["oncall_integration"]:
+        OnCallAPIClient.delete(
+            f"integrations/{integration['oncall_integration']['id']}"
+        )
+
+    # Create new integration
+    payload = {
+        "name": integration["name"],
+        "type": integration["oncall_type"],
+        "team_id": None,
```

Comment:
Just to be sure, teams are not migrated, right? (so users and resources are not really linked to a team after migration, except by the "name convention")

Reply:
@matiasb correct. at the moment none of the migrator script variants migrate teams due to the sync (e.g. you create the Grafana team, but the team isn't immediately available in OnCall).

in a future PR we might be able to implement this by:
1. modifying the public API to trigger a synchronous team sync by looking at the newly added migrator User-Agent header (similar to the terraform behaviour)
2. we would also need to then fetch the list of teams and build a map of OnCall -> <PD|Splunk|OpsGenie> team, and pass that into the migrate functions which reference teams, to know which OnCall team to assign to the newly created objects